tests/returntypes/test_adjoint_diff_new.py

```diff
@@ -0,0 +1,365 @@
+        tapes, fn = qml.gradients.finite_diff(tape)
+        numeric_val = fn(qml.execute(tapes, dev, None))
+        assert np.allclose(calculated_val, numeric_val, atol=tol, rtol=0)
+
+    @pytest.mark.autograd
+    @pytest.mark.parametrize("theta", np.linspace(-2 * np.pi, 2 * np.pi, 7))
+    def test_Rot_gradient(self, theta, tol, dev):
+        """Tests that the device gradient of an arbitrary Euler-angle-parameterized gate is
+        correct."""
+        params = np.array([theta, theta**3, np.sqrt(2) * theta])
+
+        with qml.tape.QuantumTape() as tape:
+            qml.QubitStateVector(np.array([1.0, -1.0], requires_grad=False) / np.sqrt(2), wires=0)
+            qml.Rot(*params, wires=[0])
+            qml.expval(qml.PauliZ(0))
+
+        tape.trainable_params = {1, 2, 3}
+
+        calculated_val = dev.adjoint_jacobian_new(tape)
+
+        # compare to finite differences
+        tapes, fn = qml.gradients.finite_diff(tape)
+        numeric_val = fn(qml.execute(tapes, dev, None))
+        assert np.allclose(calculated_val, numeric_val, atol=tol, rtol=0)
+
+    def test_ry_gradient(self, tol, dev):
+        """Test that the gradient of the RY gate matches the exact analytic formula."""
+
+        par = 0.23
+
+        with qml.tape.QuantumTape() as tape:
+            qml.RY(par, wires=[0])
+            qml.expval(qml.PauliX(0))
+
+        tape.trainable_params = {0}
+
+        # gradients
+        exact = np.cos(par)
+        tapes, fn = qml.gradients.finite_diff(tape)
+        grad_F = fn(qml.execute(tapes, dev, None))
+        grad_A = dev.adjoint_jacobian_new(tape)
+
+        # different methods must agree
+        assert np.allclose(grad_F, exact, atol=tol, rtol=0)
+        assert np.allclose(grad_A, exact, atol=tol, rtol=0)
+
+    def test_rx_gradient(self, tol, dev):
+        """Test that the gradient of the RX gate matches the known formula."""
+        a = 0.7418
+
+        with qml.tape.QuantumTape() as tape:
+            qml.RX(a, wires=0)
+            qml.expval(qml.PauliZ(0))
+
+        # circuit jacobians
+        dev_jacobian = dev.adjoint_jacobian_new(tape)
+        expected_jacobian = -np.sin(a)
+        assert np.allclose(dev_jacobian, expected_jacobian, atol=tol, rtol=0)
+
+    def test_multiple_rx_gradient(self, tol):
+        """Tests that the gradient of multiple RX gates in a circuit yields the correct result."""
+        dev = qml.device("default.qubit", wires=3)
+        params = np.array([np.pi, np.pi / 2, np.pi / 3])
+
+        with qml.tape.QuantumTape() as tape:
+            qml.RX(params[0], wires=0)
+            qml.RX(params[1], wires=1)
+            qml.RX(params[2], wires=2)
+
+            for idx in range(3):
+                qml.expval(qml.PauliZ(idx))
+
+        # circuit jacobians
+        dev_jacobian = dev.adjoint_jacobian_new(tape)
+        expected_jacobian = -np.diag(np.sin(params))
+        assert np.allclose(dev_jacobian, expected_jacobian, atol=tol, rtol=0)

```

Comment:
Let's check that `dev_jacobian` is a `tuple` and that each element within is an array of expected shapes. This will be relevant for other test cases too.